llvm/include/llvm/CodeGen/Register.h

```diff
@@ -160,6 +160,37 @@ template <> struct DenseMapInfo<Register> {
   }
 };
 
+/// Wrapper class representing a virtual register or register unit.
+class RegisterUnit {
```

Comment:
The name is still confusing. Maybe `VRegOrUnit` and `VRU` for variable names? (Couldn't come up with anything much better.)